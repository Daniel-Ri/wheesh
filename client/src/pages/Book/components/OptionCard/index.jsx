import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import classes from './style.module.scss';

const OptionCard = ({ seatClass, price, available, isSelected, onClick }) => (
  <div
    className={`${classes.optionCard}  ${isSelected ? classes.selected : ''} ${
      available === 'None' ? classes.disabled : ''
    }`}
    onClick={onClick}
  >
    <div className={classes.price}>{price}</div>
    <div className={classes.seatClass}>{seatClass} Class</div>
    <div className={classes.available}>
      {/* eslint-disable-next-line no-nested-ternary */}
      {available === 'Available' ? (
        <FormattedMessage id="app_available" />
      ) : available === 'Few' ? (
        <FormattedMessage id="app_few" />
      ) : (
        <FormattedMessage id="app_none" />
      )}
    </div>
  </div>
);

OptionCard.propTypes = {
  seatClass: PropTypes.string,
  price: PropTypes.string,
  available: PropTypes.string,
  isSelected: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default OptionCard;
