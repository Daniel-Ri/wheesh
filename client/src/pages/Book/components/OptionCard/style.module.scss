@import '@styles/base';

.optionCard {
  border-radius: 10px;
  width: 30%;
  height: 5.5rem;
  padding: 1rem 0.5rem;
  background-color: $color-magnolia;
  border: 1px solid transparent;
  cursor: pointer;
  transition: 0.3s ease-in-out;
  
  .price {
    font-size: 90%;
  }

  .seatClass {
    color: $color-cool-gray;
    font-size: 70%;
    text-transform: capitalize;
  }

  .available {
    color: $color-cool-gray;
    font-size: 70%;
  }

  @media screen and (max-width: $screen-sm-limit) {
    height: 7rem;

    .price {
      font-size: 80%;
    }
  }
}

.optionCard.selected {
  background-color: $color-white;
  border-color: $color-picton-blue-dark-shade;
}

.optionCard.disabled {
  cursor: no-drop;
}