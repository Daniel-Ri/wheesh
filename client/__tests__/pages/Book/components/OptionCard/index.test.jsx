import { render, fireEvent } from '@utils/testHelper';
import OptionCard from '@pages/Book/components/OptionCard';

const mockHandleClick = jest.fn();
let wrapper;

const mockProps = {
  seatClass: 'first',
  price: 'Rp 600.000',
  available: 'Available',
  isSelected: false,
  onClick: mockHandleClick,
};

beforeEach(() => {
  wrapper = render(<OptionCard {...mockProps} />);
});

describe('Component OptionCard of Book Page', () => {
  test('Renders OptionCard component', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('OptionCard')).toBeInTheDocument();
  });

  test('Calls onClick when button is clicked', () => {
    const { getByTestId } = wrapper;

    const card = getByTestId('OptionCard');
    fireEvent.click(card);

    expect(mockHandleClick).toHaveBeenCalledTimes(1);
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
