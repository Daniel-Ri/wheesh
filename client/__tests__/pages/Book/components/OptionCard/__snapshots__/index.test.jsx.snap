// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component OptionCard of Book Page Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="optionCard   "
        data-testid="OptionCard"
      >
        <div
          class="price"
        >
          Rp 600.000
        </div>
        <div
          class="seatClass"
        >
          first
           Class
        </div>
        <div
          class="available"
        >
          Tersedia
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="optionCard   "
      data-testid="OptionCard"
    >
      <div
        class="price"
      >
        Rp 600.000
      </div>
      <div
        class="seatClass"
      >
        first
         Class
      </div>
      <div
        class="available"
      >
        Tersedia
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
